_posts/2025-10-14-Scaling-Sparse-Vector-Search-to-Billions-Neural-Sparse-Now-Supports-Approximate-Search-for-Ultra-Fast-Retrieval.markdown

```diff
@@ -0,0 +1,267 @@
+
+Traditional search methods suffer from increasing query latency as collections grow. This degradation in query throughput can significantly impact user experience. In OpenSearch 2.15, we took an important first step toward addressing this challenge by introducing the [two-phase search processor](https://opensearch.org/blog/introducing-a-neural-sparse-two-phase-algorithm/). By dynamically pruning tokens with negligible weights, this approach reduces the computational load while preserving search relevance.
+
+With this same goal of efficiency and high-quality retrieval, we recently introduced sparse approximate nearest neighbor (ANN) search based on the [SEISMIC](https://dl.acm.org/doi/10.1145/3626772.3657769) (**S**pilled Clust**e**ring of **I**nverted Lists with **S**ummaries for **M**aximum **I**nner Produ**c**t Search) approximate retrieval algorithm for sparse vectors. This algorithm fundamentally changes what's possible in large-scale search. Today, we're excited to announce that the SEISMIC algorithm is now available in OpenSearch 3.3. This algorithm achieves faster query latency than traditional BM25 while preserving the semantic understanding of neural sparse models, representing a fundamental shift in the search performance landscape.
+
+## The SEISMIC algorithm
+
+A SEISMIC index consists of two components: an inverted index and a forward index. The forward index maps each document ID to its sparse embedding, where each embedding component represents a token ID and its corresponding weight. The inverted index, on the other hand, applies multiple pruning techniques to optimize efficiency and reduce storage:
+
+1. **Clustered posting list**: For each term in the inverted index, the SEISMIC algorithm sorts documents by token weights, retains only the top documents, and applies clustering in order to group similar documents.
+2. **Sparse vector summarization**: Each cluster maintains a summary vector containing only the highest-weighted tokens, enabling efficient pruning during query time.
+3. **Query-time pruning**: During query execution, the SEISMIC algorithm employs token-level and cluster-level pruning to dramatically reduce the number of documents that need scoring.
+
+The data organization in the forward and inverted indexes is presented in the following image.
+
+![Seismic Data Structure](/assets/media/blog-images/2025-10-14-Scaling-Sparse-Vector-Search-to-Billions-Neural-Sparse-Now-Supports-Approximate-Search-for-Ultra-Fast-Retrieval/seismic.png)
+
+## Try it out
+
+To try SEISMIC-based ANN sparse search, follow these steps.
+
+### Step 1: Create an index
+
+Create a sparse index by setting `index.sparse` to `true` and define a `sparse_vector` field in the index mapping:
+
+```json
+PUT sparse-vector-index
+{
+  "settings": {
+    "index": {
+      "sparse": true
+    }
+  },
+    "mappings": {
```

Comment:
@yuye-aws The mappings were nested inside the settings. I fixed the request, but please format it so it has correct indentation. Thanks.